fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -164,6 +177,232 @@ private boolean isMVApplicable(OptExpression mvExpression,
+                return  false;
+            }
+
+            List<ColumnRefOperator> leftJoinColumnRefs =
+                    leftColumnRefSet.get().getColumnRefOperators(materializationContext.getQueryRefFactory());
+            joinColumnRefs.set(0, leftJoinColumnRefs);
+        } else if (mvJoinType.isFullOuterJoin() && queryJoinType.isRightOuterJoin()) {
+            Optional<ColumnRefSet> rightColumnRefSet = usedColumnsToTable.keySet()
+                    .stream().filter(columnSet -> rightColumns.containsAll(columnSet)).findFirst();
+            if (!rightColumnRefSet.isPresent()) {
+                return false;
+            }
+            Table rightTable = usedColumnsToTable.get(rightColumnRefSet.get());
+            if (rightTable.getColumns().stream().allMatch(column -> column.isAllowNull())) {
+                // should have at least one nullable column
+                return  false;
+            }
+            List<ColumnRefOperator> rightJoinColumnRefs =
+                    rightColumnRefSet.get().getColumnRefOperators(materializationContext.getQueryRefFactory());
+            joinColumnRefs.set(1, rightJoinColumnRefs);
+        } else if (mvJoinType.isFullOuterJoin() && queryJoinType.isInnerJoin()) {
+            Optional<ColumnRefSet> leftColumnRefSet = usedColumnsToTable.keySet()
+                    .stream().filter(columnSet -> leftColumns.containsAll(columnSet)).findFirst();
+            if (!leftColumnRefSet.isPresent()) {
+                return false;
+            }
+            Table leftTable = usedColumnsToTable.get(leftColumnRefSet.get());
+            if (leftTable.getColumns().stream().allMatch(column -> column.isAllowNull())) {
+                // should have at least one nullable column
+                return  false;
+            }
+            List<ColumnRefOperator> leftJoinColumnRefs =
+                    leftColumnRefSet.get().getColumnRefOperators(materializationContext.getQueryRefFactory());
+            joinColumnRefs.set(0, leftJoinColumnRefs);
+
+            Optional<ColumnRefSet> rightColumnRefSet = usedColumnsToTable.keySet()
+                    .stream().filter(columnSet -> rightColumns.containsAll(columnSet)).findFirst();
+            if (!rightColumnRefSet.isPresent()) {
+                return false;
+            }
+            Table rightTable = usedColumnsToTable.get(rightColumnRefSet.get());
+            if (rightTable.getColumns().stream().allMatch(column -> column.isAllowNull())) {
+                // should have at least one nullable column
+                return  false;
+            }
+            List<ColumnRefOperator> rightJoinColumnRefs =
+                    rightColumnRefSet.get().getColumnRefOperators(materializationContext.getQueryRefFactory());
+            joinColumnRefs.set(1, rightJoinColumnRefs);
+        }
+        return isCompatible;
+    }
+
+    private boolean isUniqueColumns(Table table, List<String> columnNames) {
+        if (table.isOlapTable()) {
```

Comment:
what if CLOUD_NATIVE Table, and MV?

Reply:
now mv can not be primary key or unique key. add check for cloud native table